File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/2017.cs
using System;

class URI
{
    static void Main() {
        string ori = Console.ReadLine();
            
        int i, k = int.Parse(Console.ReadLine());
        int ig = 0, pos = 0;
            
        for(i = 0; i < 5; i++){
            
            string f = Console.ReadLine();
            
            int temp = 0;
            for(int j = 0; j < f.Length; j++){
                if(f[j] == ori[j]){
                    temp += 1;
                }
            }
            
            if(temp > ig){
                ig = temp;
                pos = i+1;
            }
            
        }
        
        if(ori.Length - ig > k) Console.Write("-1\n");
        else{
            Console.Write("{0}\n", pos);
            Console.Write("{0}\n", ori.Length - ig);
        }
        
    }
}

File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/2017.rb
ori = gets.chomp()
k = gets.to_i

ig = 0
pos = 0

for i in 1..5 do
    temp = 0
    f = gets.chomp()
    for j in 0..(f.length-1) do
        if(f[j] == ori[j]) then
            temp += 1
        end
    end
            
    if(temp > ig) then
        ig = temp
        pos = i
    end
end

if(ori.length - ig > k) then puts(-1)
else
    puts(pos);
    puts(ori.length - ig)
end
